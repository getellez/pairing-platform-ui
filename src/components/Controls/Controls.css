.Control {
  display: flex;
  justify-content: center;
  width: 100%;
  position: sticky;
  top: 0;
  z-index: 1;
}

.Control__action {
  margin: 0 5px;
  padding: 10px 25px;
  border-radius: 3px;
  cursor: pointer;
}

@media (min-width: 1024px) {
  .Control {
    display: flex;
    justify-content: end;
  }
}