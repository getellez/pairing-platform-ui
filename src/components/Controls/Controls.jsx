import { useNavigate } from 'react-router-dom';
import { urls } from '../../config/urls';

export const Controls = () => {
  
  const navigate = useNavigate()

  const handleLogout = (e) => {
    localStorage.removeItem('pairing-token')
    navigate(urls.loginPage)
    window.location.reload()
  }

  return (
    <section className='control'>
        {/* <div className='control__action'>
          <p>Pair</p>
        </div>
        <div className='control__action'>
          <p>Save</p>
        </div> */}
        <div className='control__action'>
          <p onClick={handleLogout}>Logout</p>
        </div>
    </section>
  )
}
