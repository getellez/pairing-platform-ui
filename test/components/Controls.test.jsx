import { render, screen } from "@testing-library/react";
import { Controls } from "../../src/components/Controls/Controls";

describe('<Controlss /> component', () => {
  test('should match the snapshot', () => {
    const { container } = render(<Controls />)
    expect( container ).toMatchSnapshot()
  });
  test('should there be 3 buttons', () => {
    render(<Controls />)
    expect(screen.getByText('Pair')).toBeTruthy()
    expect(screen.getByText('Save')).toBeTruthy()
    expect(screen.getByText('Cancel')).toBeTruthy()
  })
})