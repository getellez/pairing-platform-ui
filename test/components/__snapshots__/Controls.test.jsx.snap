// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controlss /> component should match the snapshot 1`] = `
<div>
  <section
    class="control"
  >
    <div
      class="control__action"
    >
      <p>
        Pair
      </p>
    </div>
    <div
      class="control__action"
    >
      <p>
        Save
      </p>
    </div>
    <div
      class="control__action"
    >
      <p>
        Cancel
      </p>
    </div>
  </section>
</div>
`;
